plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/PredicatePushdownController.java

```diff
@@ -62,6 +67,18 @@ public interface PredicatePushdownController
         return new DomainPushdownResult(simplifiedDomain, domain);
     };
 
+    static PredicatePushdownController pushdownDiscreteValues(Type type)
+    {
+        return (session, domain) -> {
+            Optional<Collection<Object>> expandedRange = domain.getValues().tryExpandRanges(getDomainCompactionThreshold(session));
+            if (expandedRange.isPresent() && expandedRange.get().size() > 1) {
```

Comment:
Why do we need this `&& expandedRange.get().size() > 1` condition ?
It looks unnecessary to me

Reply:
There were tests failures without that condition such https://github.com/trinodb/trino/actions/runs/11081955893/job/30794374722?pr=23598#step:19:7989 but https://github.com/trinodb/trino/pull/23417#discussion_r1790304736 helped fix this one. Let's see if there are other failures.